src/monty/io.py

```diff
@@ -104,68 +157,71 @@ def reverse_readline(
 
     Args:
         m_file (File): File stream to read (backwards)
-        blk_size (int): The buffer size. Defaults to 4096.
+        blk_size (int): The buffer size in bytes. Defaults to 4096.
         max_mem (int): The maximum amount of memory to involve in this
             operation. This is used to determine when to reverse a file
             in-memory versus seeking portions of a file. For bz2 files,
             this sets the maximum block size.
 
-    Returns:
-        Generator that yields lines from the file. Behave similarly to the
-        file.readline() function, except the lines are returned from the back
-        of the file.
+    Yields:
+        Lines from the file. Behave similarly to the file.readline function,
+        except the lines are returned from the back of the file.
     """
-    # Check if the file stream is a bit stream or not
+    # Generate line ending
+    l_end = _get_line_ending(m_file)
+
+    # Check if the file stream is a buffered text stream
     is_text = isinstance(m_file, io.TextIOWrapper)
 
     try:
         file_size = os.path.getsize(m_file.name)
     except AttributeError:
-        # Bz2 files do not have name attribute. Just set file_size to above
-        # max_mem for now.
+        # Bz2 files do not have name attribute.
+        # Just set file_size to max_mem for now.
         file_size = max_mem + 1
 
-    # If the file size is within our desired RAM use, just reverse it in memory
-    # GZip files must use this method because there is no way to negative seek
+    # If the file size is within desired RAM limit, just reverse it in memory.
+    # GZip files must use this method because there is no way to negative seek.
     # For windows, we also read the whole file.
     if file_size < max_mem or isinstance(m_file, gzip.GzipFile) or os.name == "nt":
         for line in reversed(m_file.readlines()):
-            yield line.rstrip()
```

Comment:
I believe this causes different behavior in Windows and Linux, in Windows (`os.name == "nt"`) the line is right-stripped but not in Linux. 

I think we should NOT preprocess data (`rstrip`) within both `reverse_readfile` and `reverse_readline` because we want it to behave similar to the build in `readlines` just backwards (`readlines` would not `rstrip` lines): https://github.com/materialsvirtuallab/monty/blob/1270c7b230b43cdbc4f5f184ac645ece31d49b61/src/monty/io.py#L114-L116

As such we should treat lines **as is**, just backwards.